src/VehicleSelect/VehicleSelect.spec.tsx

```diff
@@ -0,0 +1,194 @@
+    await expect(page.getByText("onChange").nth(3)).toBeVisible();
+    await expect(page.getByText("onChange").nth(4)).toBeVisible();
+    await expect(page.getByText("onChange").nth(5)).not.toBeVisible();
+    // Open the onChange description and assert the expected value
+    await page.getByText("onChange").first().click();
+    await page.getByText("0").first().click();
+    await page.getByText("1", { exact: true }).first().click();
+    await expect(page.getByText('"911"').first()).toBeVisible();
+    await expect(page.getByText('"2015"').first()).toBeVisible();
+    await expect(
+      page.getByText('"64c8c4cccc8d6f00130b367e"').first()
+    ).toBeVisible();
+    await expect(
+      page.getByText('"64c8c4cccc8d6f00130b366b"').first()
+    ).toBeVisible();
+    await expect(page.getByText('"Gate 1"').first()).toBeVisible();
+    await expect(page.getByText('"Gate 1"').nth(1)).toBeVisible();
+    // Open the gate dropdown and select the second option then click away
+    await page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByLabel("Gate *")
+      .click();
+    await page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByRole("option", { name: "Gate 2" })
+      .getByRole("checkbox")
+      .check();
+    await page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByLabel("Close")
+      .click();
+    // Assert that onChange is called five times
+    await expect(page.getByText("onChange").first()).toBeVisible();
+    await expect(page.getByText("onChange").nth(1)).toBeVisible();
+    await expect(page.getByText("onChange").nth(2)).toBeVisible();
+    await expect(page.getByText("onChange").nth(3)).toBeVisible();
+    await expect(page.getByText("onChange").nth(4)).toBeVisible();
+    await expect(page.getByText("onChange").nth(5)).toBeVisible();
+    await expect(page.getByText("onChange").nth(6)).not.toBeVisible();
+    // Open the onChange description and assert the expected value
+    await page.getByText("onChange").first().click();
+    await page.getByText("0", { exact: true }).first().click();
+    await page.getByText("1", { exact: true }).first().click();
+    await page.getByText("2", { exact: true }).click();
+    await page.getByText("3", { exact: true }).click();
+    await expect(page.getByText('"911"').first()).toBeVisible();
+    await expect(page.getByText('"2015"').first()).toBeVisible();
+    await expect(
+      page.getByText('"64c8c4cccc8d6f00130b367e"').first()
+    ).toBeVisible();
+    await expect(
+      page.getByText('"64c8c4cccc8d6f00130b366b"').first()
+    ).toBeVisible();
+    await expect(page.getByText('"Gate 1"').first()).toBeVisible();
+    await expect(page.getByText('"Gate 1"').nth(1)).toBeVisible();
+    await expect(page.getByText('"Gate 2"').first()).toBeVisible();
+    await expect(page.getByText('"Gate 2"').nth(1)).toBeVisible();
+  });
+});
```

Comment:
I personally find the tests a bit difficult to follow, I would suggest maybe set up an arrays containing the value we expect for each option. And then loop with the array performing each check maybe. Maybe check with @dmbarry86 on his thoughts, below is an example..

```
import { expect, test } from "@playwright/test";

const selectors = [
  { label: "Project Code *", option: "911" },
  { label: "Model Year *", option: "2015" },
  { label: "Vehicle Variant *", option: "JS - 3.6 l6 - 397kW - 7MT -" },
  { label: "Vehicle Variant *", option: "MP - 3.6 l6 - 397kW - 7MT -" },
  { label: "Gate *", option: "Gate 1" },
  { label: "Gate *", option: "Gate 2" },
];

test.describe("Vehicle Select tests", () => {
  test("Test onChange is called with the expected values", async ({ page }) => {
    await page.goto(
      "http://localhost:6006/?path=/story/selectors-vehicleselect--default"
    );

    for (let i = 0; i < selectors.length; i++) {
      const { label, option } = selectors[i];

      // Open dropdown and select option
      await page
        .frameLocator('iframe[title="storybook-preview-iframe"]')
        .getByLabel(label)
        .click();
      await page
        .frameLocator('iframe[title="storybook-preview-iframe"]')
        .getByRole("option", { name: option })
        .click();

      // Check onChange calls
      await page.getByRole("tab", { name: "Actions 1" }).click();
      await expect(page.getByText("onChange")).toBeVisible();
      await expect(page.getByText(`onChange ${i + 1}`)).toBeVisible();
      await expect(page.getByText(`onChange ${i + 2}`)).not.toBeVisible();

      // Check onChange value
      await page.getByText("onChange").click();
      await page.getByText(i.toString()).click();
      await expect(page.getByText(`"${option}"`)).toBeVisible();
    }
  });
});
```

Reply:
I agree, it's very hard to follow. Also not helped by the fact there are no newlines between each code block so it's just one big blur. The suggestion from Syed is much easier to read. There's a tradeoff here between what is automatically generated by the Playwright test generator (presumably what Victor used) vs the optimised readable code a developer would write.

Reply:
Thanks for the feedback, I have addressed this and have submitted some changes. 
I based by implementation off your suggestion and made some modifications to it and the tests are passing, however I must highlight per each of the loops that it is doing it checks that the last value was added only, not that the other values remain.

Reply:
I would suggest that we also add a check after the for loop to check at the selected value in those field are as expected.

Reply:
I've added this